Assets/Scripts/SS3D/Systems/Inventory/Containers/Hand.cs

```diff
@@ -0,0 +1,145 @@
+﻿using SS3D.Systems.Inventory.Containers;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using SS3D.Systems.Inventory.Items;
+using System.Linq;
+using SS3D.Interactions.Interfaces;
+using SS3D.Interactions;
+using FishNet.Object;
+using static SS3D.Systems.Inventory.Containers.AttachedContainer;
+
+namespace SS3D.Systems.Inventory.Containers
+{
+	/// <summary>
+	/// A hand is what an entity uses to grab and hold items, to interact with things in range.
+	/// </summary>
+	public class Hand : InteractionSource, IInteractionRangeLimit, IInteractionOriginProvider
+	{
+		/// <summary>
+		/// Container linked to this hand, necessary to hold stuff.
+		/// </summary>
+		public AttachedContainer Container;
+
+		/// <summary>
+		/// Horizontal and vertical max distance to interact with stuff.
+		/// </summary>
+		public RangeLimit range = new(1.5f, 2);
+
+		// pickup icon that this hand uses when there's a pickup interaction
+		// TODO: When AssetData is on, we should update this to not use this
+		public Sprite pickupIcon;
+
+		/// <summary>
+		/// The item held in this hand, if it exists
+		/// </summary>
+		public Item ItemInHand => Container.Items.FirstOrDefault();
+
+		/// <summary>
+		/// Point from where distances for interaction is computed.
+		/// </summary>
+		public Transform interactionOrigin;
+
+		/// <summary>
+		/// the hands script controlling this hand.
+		/// </summary>
+		public Hands handsController;
+
+		public Vector3 InteractionOrigin => interactionOrigin.position;
+
+		public delegate void HandHandler(Hand hand);
+		public event HandHandler OnHandDisabled;
```

Comment:
Fix naming of the fields

Reply:
not sure what you expect, I tried a renaming, you'll tell me if that works

Reply:
Change range to Range, pickupIcon to PickupIcon, interactionOrigin to InteractionOrigin, handsController to HandsController